OrbitQt/servicedeploymanager.cpp

```diff
@@ -199,6 +222,15 @@ outcome::result<void> ServiceDeployManager::CopyOrbitServicePackage() {
 
 ErrorMessageOr<void> ServiceDeployManager::CopyFileToLocal(std::string_view source,
                                                            std::string_view destination) {
+  if (QThread::currentThread() != thread()) {
+    ErrorMessageOr<void> result = outcome::success();
+    DeferToBackgroundThreadAndWait(
+        this, [&, source = std::string{source}, destination = std::string{destination}]() {
+          result = CopyFileToLocal(source, destination);
```

Comment:
Maybe move this to another function and have a CHECK(QThread::currentThread() != thread()) there?

In which case you won't need this if here.

Reply:
Same for Cancel

Reply:
I restructured this entirely and moved the thread into ServiceDeployManager which means it now always uses a background thread.